terraform-provider-constellation/internal/provider/cluster_resource.go

```diff
@@ -0,0 +1,308 @@
+type ClusterResourceModel struct {
+	UID                   types.String `tfsdk:"uid"`
+	Name                  types.String `tfsdk:"name"`
+	Image                 types.String `tfsdk:"image"`
+	KubernetesVersion     types.String `tfsdk:"kubernetes_version"`
+	Debug                 types.Bool   `tfsdk:"debug"`
+	InitEndpoint          types.String `tfsdk:"init_endpoint"`
+	KubernetesAPIEndpoint types.String `tfsdk:"kubernetes_api_endpoint"`
+	MicroserviceVersion   types.String `tfsdk:"constellation_microservices_version"`
+	ExtraMicroservices    types.Object `tfsdk:"extra_microservices"`
+	MasterSecret          types.String `tfsdk:"master_secret"`
+	InitSecret            types.String `tfsdk:"init_secret"`
+	Attestation           types.Object `tfsdk:"attestation"`
+	OwnerID               types.String `tfsdk:"owner_id"`
+	ClusterID             types.String `tfsdk:"cluster_id"`
+	Kubeconfig            types.String `tfsdk:"kubeconfig"`
+	// NetworkConfig 			 types.Object `tfsdk:"network_config"` // TODO(elchead): do when clear what is needed
+}
+
+// Metadata returns the metadata of the resource.
+func (r *ClusterResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_cluster"
+}
+
+// Schema returns the schema of the resource.
+func (r *ClusterResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Resource for a Constellation cluster.",
+		Description:         "Resource for a Constellation cluster.",
+
+		Attributes: map[string]schema.Attribute{
+			"uid": schema.StringAttribute{
+				MarkdownDescription: "The UID of the cluster.",
+				Description:         "The UID of the cluster.",
+				Required:            true,
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Name used in the cluster's named resources / cluster name.",
+				Description:         "Name used in the cluster's named resources / cluster name.",
+				Optional:            true, // TODO(elchead): use "constell" as default
+			},
+			"image": schema.StringAttribute{
+				MarkdownDescription: "The Constellation OS image to use in the CSP specific reference format. Use the `constellation_image` data source to find the correct image for your CSP.",
+				Description:         "The Constellation OS image to use in the CSP specific reference format. Use the `constellation_image` data source to find the correct image for your CSP. When not set, the latest default version will be used.",
+				Optional:            true,
+			},
+			"kubernetes_version": schema.StringAttribute{
+				MarkdownDescription: "The Kubernetes version to use for the cluster.",
+				Description:         "The Kubernetes version to use for the cluster. When not set, the latest default version will be used.", // TODO(elchead): refer to supported versions; wait for https://github.com/edgelesssys/constellation/pull/2661
```

Comment:
You could just embed this information into the binary using `versions.SupportedK8sVersions`.
If Terraform uses this Schema function to generate the docs, it should always contain the correct information for a released version

Reply:
No, I was referring to `versions.SupportedK8sVersions` which returns a list of the Kubernetes versions that are supported by this version of Constellation.
You don't need to refer to our docs for this information, when you can just provide it inline.